k8s_pod_filter.go

```diff
@@ -152,8 +152,53 @@ func (f *PodFilter) ShouldTailLogs(pod *Pod) (bool, error) {
 	return false, nil
 }
 
+// A TailAllFilter reuses AnnotationPodFilter but inverts the logic - tail all pods except those with TailLogs=false
+type TailAllFilter struct {
+	*AnnotationPodFilter
+}
+
+func NewTailAllFilter(kubeHost string, kubePort int, timeout time.Duration, credsPath string) *TailAllFilter {
+	annotationFilter := NewAnnotationPodFilter(kubeHost, kubePort, timeout, credsPath)
+	if annotationFilter == nil {
+		return nil
+	}
+	return &TailAllFilter{AnnotationPodFilter: annotationFilter}
+}
+
+func (f *TailAllFilter) ShouldTailLogs(pod *Pod) (bool, error) {
+	body, err := f.makeRequest(
+		"/api/v1/namespaces/" + pod.Namespace + "/pods?limit=100000&labelSelector=ServiceName%3D" + pod.ServiceName,
+	)
+	if err != nil {
+		return false, err
+	}
+
+	var pods K8sPodsMetadata
+	err = json.Unmarshal(body, &pods)
+	if err != nil {
+		return false, fmt.Errorf("unable to decode response from K8s: %s", err)
+	}
+
+	// Default to tail if we don't know about this pod
+	if len(pods.Items) < 1 {
+		return true, nil
```

Comment:
Ideally all of this is broken out into a shared function like `getPodData` or something like that. That would make the body of these two implementations pretty small.